### PR TITLE
[alertmanager] feat: support for custom volumes and environment variables

### PR DESCRIPTION
**Pull request by kshcherban**
#### What this PR does / why we need it

This PR adds support for custom environment variables and volumes to alertmanager container.
Can be useful if you want to use secrets from [CSI secrets store](https://secrets-store-csi-driver.sigs.k8s.io/) in alertmanager configuration.

#### Which issue this PR fixes

N/A

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/alertmanager/templates/_helpers.tpl [hunk 1]
```diff
@@ -84,4 +84,18 @@ Define Pdb apiVersion
 {{- else }}
 {{- printf "policy/v1beta1" }}
 {{- end }}
-{{- end }}
\ No newline at end of file
+{{- end }}
+
+{{/*
+Renders a value that contains template.
+Based on bitnami common
```
[the remark this answers is not in the document]
Comment (kshcherban) [reply charts/alertmanager/templates/_helpers.tpl]: Added as dependency.

File: charts/alertmanager/values.yaml [hunk 2]
```diff
@@ -206,3 +206,19 @@ configmapReload:
...
+  #   mountPath: /usr/share/extras
+  #   readOnly: true
+
+## Optionally specify extra list of additional volumes
+extraVolumes: []
+  # - name: extras
+  #   emptyDir: {}
+
+## Optionally specify extra environment variables to add to alertmanager container
+extraEnvVars: []
```
[the remark this answers is not in the document]
Comment (kshcherban) [reply charts/alertmanager/values.yaml]: Renamed.

File: charts/alertmanager/templates/statefulset.yaml [hunk 3]
```diff
@@ -105,6 +105,9 @@ spec:
                 fieldRef:
                   apiVersion: v1
                   fieldPath: status.podIP
+            {{- if .Values.extraEnvVars }}
+            {{- include "alertmanager.tplvalues.render" (dict "value" .Values.extraEnvVars "context" $) | nindent 12 }}
```
[the remark this answers is not in the document]
Comment (kshcherban) [reply charts/alertmanager/templates/statefulset.yaml]: Renamed.

**Review by naseemkullah — APPROVED** [review]

**Review by naseemkullah — APPROVED** [review]

**Review by monotek — CHANGES_REQUESTED** [review]
chart version eeds to be updated.

i'm not sure if its desirable to use bitnami/common as dependency, as we have no control over it,

i had a prometheus-community/common chart in mind before, which would be used by all charts here, but would be a lot of work to build and maintain it and refactor all the charts.

File: charts/alertmanager/.gitignore [hunk 4]
```diff
@@ -0,0 +1,2 @@
+requirements.lock
```
[the remark this answers is not in the document]
Comment (kshcherban) [reply charts/alertmanager/.gitignore:1]: I copied .gitignore from [here](https://github.com/prometheus-community/helm-charts/tree/main/charts/prometheus). Thought if there are dependencies you need to add it.
Ok will remove.

File: charts/alertmanager/Chart.yaml [hunk 5]
```diff
@@ -6,7 +6,7 @@ icon: https://raw.githubusercontent.com/prometheus/prometheus.github.io/master/a
 sources:
   - https://github.com/prometheus/alertmanager
 type: application
-version: 0.22.2
+version: 0.22.3
```
Comment (monotek) [inline charts/alertmanager/Chart.yaml:9]: new feautere should raise minor version

**Review by monotek — CHANGES_REQUESTED** [review]

File: charts/alertmanager/Chart.yaml (hunk 5 above)
Comment (kshcherban) [reply charts/alertmanager/Chart.yaml]: Done.

**Review by monotek — APPROVED** [review]

**Pull request merged**